src/cljc/proton/assoc_in_after.cljc

```diff
@@ -0,0 +1,39 @@
+(ns proton.assoc-in-after
+  "Functions of `assoc-in` in a specific position")
+
+(defn- contains-in? [m ks]
+  (contains? (get-in m (butlast ks)) (last ks)))
+
+(defn- assoc-in* [m ks v]
+  (if (empty? ks)
+    v
+    (assoc-in m ks v)))
+
+(defn assoc-in-after
+  "Inserts `v` into `ks` after `before-ks` in `omap`"
+  [omap ks v before-ks]
+  (cond
+    (empty? ks) v
+    (empty? before-ks) (assoc-in* omap ks v)
+    (contains-in? omap ks) (assoc-in* omap ks v)
+    (contains? omap (first ks)) (update omap
+                                        (first ks)
+                                        assoc-in-after
+                                        (rest ks)
+                                        v
+                                        (rest before-ks))
+    (= ::first (first before-ks)) (into (array-map)
+                                        (cons [(first ks)
+                                               (assoc-in* (array-map)
```

Comment:
We should use [`ordered-map`](https://github.com/clj-commons/ordered), not `array-map`.
Because, `array-map` does not keep order if it has many records.

```
user=> (into (array-map) (mapv (fn [i] [i i]) (range 8)))
{0 0, 1 1, 2 2, 3 3, 4 4, 5 5, 6 6, 7 7}
user=> (into (array-map) (mapv (fn [i] [i i]) (range 9)))
{0 0, 7 7, 1 1, 4 4, 6 6, 3 3, 2 2, 5 5, 8 8}
```

Or, we use `apply array-map` but carefully (It is fragile...).

```
user=> (apply array-map (mapcat (fn [i] [i i]) (range 80)))
{0 0, 1 1, 2 2, 3 3, 4 4, 5 5, 6 6, 7 7, 8 8, 9 9, 10 10, 11 11, 12 12, 13 13, 14 14, 15 15, 16 16, 17 17, 18 18, 19 19, 20 20, 21 21, 22 22, 23 23, 24 24, 25 25, 26 26, 27 27, 28 28, 29 29, 30 30, 31 31, 32 32, 33 33, 34 34, 35 35, 36 36, 37 37, 38 38, 39 39, 40 40, 41 41, 42 42, 43 43, 44 44, 45 45, 46 46, 47 47, 48 48, 49 49, 50 50, 51 51, 52 52, 53 53, 54 54, 55 55, 56 56, 57 57, 58 58, 59 59, 60 60, 61 61, 62 62, 63 63, 64 64, 65 65, 66 66, 67 67, 68 68, 69 69, 70 70, 71 71, 72 72, 73 73, 74 74, 75 75, 76 76, 77 77, 78 78, 79 79}
```

Reply:
@ayamada Thank you for your review! 

> By the way, write diff-assert to support without cljs somewhere?

Yes :sob: I tried using `assert-expr` on cljs, but I gave up.

> We should use ordered-map, not array-map.

I see. However, proton's policies has `Zero dependencies`, so I didn't use `ordered-map` :thinking: 
@federkasten What are your thoughts on using [ordered-map](https://github.com/clj-commons/ordered) ?

Reply:
`Zero dependencies` is proton's important policy, so I want to avoid depending on `ordered-map`. Could you try to handle `omap` directly whether it is an array-map or an ordered-map, without using `ordered-map` library?

Reply:
OK. I try it.

Reply:
@ayamada @federkasten 
I changed `assoc-in-after` to be independent of `array-map`. 
 https://github.com/xcoo/proton/pull/14/commits/510e9d03eadac534289a5a69f4d42eb2abe11139
Functions such as `into` execute on empty elements of `omap`. So if you want to keep the order of large `omap`, you can pass the `ordered-map` to `omap`.
proton is independent of `ordered-map`, but you can use it at the argument.